src/Datadog.Trace/Sampling/RuleBasedSampler.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Sampling
+{
+    internal class RuleBasedSampler : ISampler
+    {
+        private const ulong KnuthFactor = 1_111_111_111_111_111_111;
+
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RuleBasedSampler>();
+
+        private readonly IRateLimiter _limiter;
+        private readonly List<ISamplingRule> _rules = new List<ISamplingRule>();
+
+        private readonly object _sampleRateGate = new object();
+
+        private Dictionary<string, float> _sampleRates = new Dictionary<string, float>();
+
+        public RuleBasedSampler(IRateLimiter limiter)
+        {
+            _limiter = limiter ?? new RateLimiter(null);
+        }
+
+        public void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampleRates)
```

Comment:
`IEnumerable<KeyValuePair<string, float>>` doesn't have a restriction on duplicate keys, but the `Dictionary<string, float>` below does. Should the argument be `IDictionary<string, float>` instead?

Reply:
This is carry over from the deleted RateByServiceSampler. Happy to change it though.

Reply:
Haha, ok. So is this a case of me ~reviewing~ nitpicking code I wrote a few months ago? 🤦‍♂ 
I'll recuse myself. Carry on.